### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by BSvystun2707**

### CODE REVIEW

File: src/checkPassword.test.js [hunk 1]
```diff
@@ -8,12 +8,37 @@ describe(`Function 'checkPassword':`, () => {
...
 
   it(`should return boolean`, () => {
+    const result = checkPassword('Password1!');
 
+    expect(typeof result).toBe('boolean');
   });
 
   it(`should return 'true' for the valid password with 8 characters`, () => {
+    expect(checkPassword('Password1!')).toBeTruthy();
+  });
```
Comment (4ksandrey) [inline src/checkPassword.test.js]: Here isn't 8 chars) 10 chars including 1 capital, 1 spec and 1 number

File: src/checkPassword.test.js [hunk 2]
```diff
@@ -8,12 +8,37 @@ describe(`Function 'checkPassword':`, () => {
...
+
+  it(`should return 'false' for the invalid password
+  with cyrrilic chars`, () => {
+    expect(checkPassword('Пасворд1!')).toBeFalsy();
+  });
+
+  it(`should return 'false' for 
+  the invalid password with less then 8 characters`, () => {
+    expect(checkPassword('Qwert!1')).toBeFalsy();
+  });
```
Comment (4ksandrey) [inline src/checkPassword.test.js]: Don't forget about boundary values! 8-16, so your tests should include verification for 7, 8, 16, 17 chars including all required symbols, fix it

File: src/checkPassword.test.js [hunk 3]
```diff
@@ -8,12 +8,37 @@ describe(`Function 'checkPassword':`, () => {
   });
 
   it(`should return boolean`, () => {
+    const result = checkPassword('Password1!');
 
+    expect(typeof result).toBe('boolean');
   });
```
Comment (4ksandrey) [inline src/checkPassword.test.js]: What about an empty input and input with spaces?

**Review by 4ksandrey — CHANGES_REQUESTED** [review]
Check the comments

File: src/checkPassword.test.js [hunk 4]
```diff
@@ -8,12 +8,58 @@ describe(`Function 'checkPassword':`, () => {
...
+   empty braces`, () => {
+    const res = checkPassword('');
+
+    expect(res).toBeFalsy();
+  });
+
+  it(`should return "false" for input with
+   empty spaces in braces`, () => {
+    expect(checkPassword(' ')).toBeFalsy();
   });
```
Comment (4ksandrey) [inline src/checkPassword.test.js]: Input should be some string with valid length including spaces

File: src/checkPassword.test.js [hunk 5]
```diff
@@ -8,12 +8,58 @@ describe(`Function 'checkPassword':`, () => {
...
+
+  it(`should return 'false' for the invalid password
+   with lower case chars`, () => {
+    expect(checkPassword('qwert!1')).toBeFalsy();
+  });
+
+  it(`should return 'false' for the invalid password
+   with 16+ chars`, () => {
+    expect(checkPassword('Qwerty!12345678901')).toBeFalsy();
+  });
```
Comment (4ksandrey) [inline src/checkPassword.test.js]: This test is duplicate, so remove it

File: src/checkPassword.test.js [hunk 6]
```diff
@@ -8,12 +8,58 @@ describe(`Function 'checkPassword':`, () => {
...
+
+  it(`should return 'false' for the invalid 
+  password without digit chars`, () => {
+    expect(checkPassword('Str@ng')).toBeFalsy();
+  });
+
+  it(`should return 'false' for the invalid password
+   with lower case chars`, () => {
+    expect(checkPassword('qwert!1')).toBeFalsy();
+  });
```
Comment (4ksandrey) [inline src/checkPassword.test.js:59]: So your have verified input without number and capital letter, ok, but what about input without spec symbol?

**Review by 4ksandrey — CHANGES_REQUESTED** [review]
Check the comments

**Review by 4ksandrey — APPROVED** [review]
Accepted, well done, but next time it would be better to put space in the middle of the string)) like: "some pa$$1"
